app/src/main/java/com/dcrandroid/activities/privacy/SetupPrivacy.kt

```diff
@@ -25,14 +34,84 @@ class SetupPrivacy : BaseActivity() {
         wallet_name.text = wallet.name
 
         btn_setup_mixer.setOnClickListener {
-            val intent = Intent(this, SetupMixerAccounts::class.java)
-            intent.putExtra(Constants.WALLET_ID, wallet.id)
-            finish()
-            startActivity(intent)
+            //Check for account name "imported" and subtract 1 from total account count
+            if (wallet.accounts.toString().contains("imported", ignoreCase = true)) {
+                val walletCount = wallet.accountsRaw.count - 1
+                if (walletCount > 1) {
+                    val intent = Intent(this, SetupMixerAccounts::class.java)
+                    intent.putExtra(Constants.WALLET_ID, wallet.id)
+                    finish()
+                    startActivity(intent)
+                } else {
+                    InfoDialog(this)
+                        .setDialogTitle(getString(R.string.privacy_intro_dialog_title))
+                        .setMessage(Html.fromHtml(getString(R.string.privacy_intro_dialog_desc)))
+                        .setNegativeButton(getString(R.string.cancel))
+                        .setPositiveButton(
+                            getString(R.string.begin_setup)
+                        ) { _, _ ->
+                            checkAccountNameConflict()
+                        }
+                        .show()
+                }
+            }
         }
 
         go_back.setOnClickListener { finish() }
 
         multiWallet!!.setBoolConfigValueForKey(Constants.CHECKED_PRIVACY_PAGE, true)
     }
+
+    fun checkAccountNameConflict() {
```

Comment:
i see you changed the functions from a private function, what i meant was to make it a global function, since this is the second instance of this function